Example/Frameworks/Database/Logger.swift

```diff
@@ -27,35 +27,133 @@ import UIKit
 import Willow
 
 extension LogLevel {
-    /// Custom log level for SQL log messages with a bitmask of `1 << 8`.
-    public static let sql = LogLevel(rawValue: 0b00000000_00000000_00000001_00000000)
+    // Namespace our custom levels
+    public enum Database {
+        /// Custom log level for SQL log messages with a bitmask of `1 << 8`.
+        public static var sql = LogLevel(rawValue: 0b00000000_00000000_00000001_00000000)
+    }
 }
 
 // MARK: -
 
+// Extend Logger to have `sql` log level functions.
 extension Logger {
-    func sql(_ message: @autoclosure @escaping () -> String) {
-        logMessage(message, with: .sql)
+    func sql(_ message: @autoclosure @escaping () -> Message) {
+        logMessage(message, with: LogLevel.Database.sql)
     }
```

Comment:
Yeah, calling `log.Database.sql("message")` is crazy. We definitely don't want that. But if you made the `sql` function `public`, you'd need to do something. I guess my end point is that it's important to expose custom `LogLevel` types as `public` so you can override their bitmask, but equally important to keep your custom convenience functions `internal` so you don't have to deal with collisions with other frameworks due to lack of namespacing.